io/zenoh-transport/src/common/pipeline.rs

```diff
@@ -795,22 +764,14 @@ impl TransmissionPipelineProducer {
 
         // If message is droppable, compute a deadline after which the sample could be dropped
         let (wait_time, max_wait_time) = if msg.is_droppable() {
-            // Checked if we are blocked on the priority queue and we drop directly the message
-            if self.status.is_congested(priority) {
```

Comment:
By removing this check, the code will always block in the `zlock!(self.stage_in[idx])` at line `773` when congestion happens. I think the general idea is to avoid getting the lock if we are already in the congested scenario.

Reply:
It will not block for long, because if the pipeline is congested, no batch will be available and `push_network_message` will return directly.
Also, having the shortcut before the lock implies that each other messages waiting for the lock will have to wait the deadline while we know that the pipeline is congested.
And last, this is easier to synchronize properly, and touch the least possible amount of code.

Reply:
You will block for long because if the previous message is reliable, it will wait for a new batch. So it won't be possible to do an early drop in case of best effort message.

Reply:
That's a valid argument.